<!DOCTYPE html>
<html lang="en" dir="ltr">
    <head><meta charset='utf-8'>
<meta name='viewport' content='width=device-width, initial-scale=1'><meta name='description' content="(Image from gnomAD website)\n計畫簡介 The Genome Aggregation Database (gnomAD) 前身為 2014 年發起之 Exome Aggregation Consortium (ExAC)，計畫主旨為收集外顯子定序資料(WES)進行大規模樣本分析；而 gnomAD 資料庫不僅涵蓋外顯子資料(WES)也致力於收集全基因定序(WGS)進行族群性分析，本文將介紹 gnomAD v3, v2 版的樣本收集與分析流程，並閱讀2020 年發表之 gnomAD SV 文章。\n樣本收集 下表整理 gnomAD 兩種版本資料庫：\nMost of the genomes from v2 are included in v3.1.\nversion reference (type)sample size gnomAD v2.1 GRCh37 (WES) 125,748 ; (WGS) 15,708 gnomAD-SV v2.1 GRCh37 (WGS) 10,847 gnomAD v3.1 GRCh38 (WGS) 76,156 資料庫的樣本由不同cohort study以及大規模基因體計畫組成，像官網提到gnomAD有部分樣本來自1000 Genome Project:\n">
<title>生資系列： gnomAD v3 database 介紹</title>

<link rel='canonical' href='https://pxchen110.github.io/p/%E7%94%9F%E8%B3%87%E7%B3%BB%E5%88%97-gnomad-v3-database-%E4%BB%8B%E7%B4%B9/'>

<link rel="stylesheet" href="/scss/style.min.663803bebe609202d5b39d848f2d7c2dc8b598a2d879efa079fa88893d29c49c.css"><meta property='og:title' content="生資系列： gnomAD v3 database 介紹">
<meta property='og:description' content="(Image from gnomAD website)\n計畫簡介 The Genome Aggregation Database (gnomAD) 前身為 2014 年發起之 Exome Aggregation Consortium (ExAC)，計畫主旨為收集外顯子定序資料(WES)進行大規模樣本分析；而 gnomAD 資料庫不僅涵蓋外顯子資料(WES)也致力於收集全基因定序(WGS)進行族群性分析，本文將介紹 gnomAD v3, v2 版的樣本收集與分析流程，並閱讀2020 年發表之 gnomAD SV 文章。\n樣本收集 下表整理 gnomAD 兩種版本資料庫：\nMost of the genomes from v2 are included in v3.1.\nversion reference (type)sample size gnomAD v2.1 GRCh37 (WES) 125,748 ; (WGS) 15,708 gnomAD-SV v2.1 GRCh37 (WGS) 10,847 gnomAD v3.1 GRCh38 (WGS) 76,156 資料庫的樣本由不同cohort study以及大規模基因體計畫組成，像官網提到gnomAD有部分樣本來自1000 Genome Project:\n">
<meta property='og:url' content='https://pxchen110.github.io/p/%E7%94%9F%E8%B3%87%E7%B3%BB%E5%88%97-gnomad-v3-database-%E4%BB%8B%E7%B4%B9/'>
<meta property='og:site_name' content='Xuan&#39;s Blog'>
<meta property='og:type' content='article'><meta property='article:section' content='Post' /><meta property='article:tag' content='Bioinformatics' /><meta property='article:tag' content='Database' /><meta property='article:published_time' content='2024-05-23T00:00:00&#43;00:00'/><meta property='article:modified_time' content='2024-05-23T00:00:00&#43;00:00'/><meta property='og:image' content='https://pxchen110.github.io/post/bioinfo-gnomad-v3/20240516_img1.png' />
<meta name="twitter:title" content="生資系列： gnomAD v3 database 介紹">
<meta name="twitter:description" content="(Image from gnomAD website)\n計畫簡介 The Genome Aggregation Database (gnomAD) 前身為 2014 年發起之 Exome Aggregation Consortium (ExAC)，計畫主旨為收集外顯子定序資料(WES)進行大規模樣本分析；而 gnomAD 資料庫不僅涵蓋外顯子資料(WES)也致力於收集全基因定序(WGS)進行族群性分析，本文將介紹 gnomAD v3, v2 版的樣本收集與分析流程，並閱讀2020 年發表之 gnomAD SV 文章。\n樣本收集 下表整理 gnomAD 兩種版本資料庫：\nMost of the genomes from v2 are included in v3.1.\nversion reference (type)sample size gnomAD v2.1 GRCh37 (WES) 125,748 ; (WGS) 15,708 gnomAD-SV v2.1 GRCh37 (WGS) 10,847 gnomAD v3.1 GRCh38 (WGS) 76,156 資料庫的樣本由不同cohort study以及大規模基因體計畫組成，像官網提到gnomAD有部分樣本來自1000 Genome Project:\n"><meta name="twitter:card" content="summary_large_image">
    <meta name="twitter:image" content='https://pxchen110.github.io/post/bioinfo-gnomad-v3/20240516_img1.png' />
    </head>
    <body class="
    article-page
    ">
    <script>
        (function() {
            const colorSchemeKey = 'StackColorScheme';
            if(!localStorage.getItem(colorSchemeKey)){
                localStorage.setItem(colorSchemeKey, "auto");
            }
        })();
    </script><script>
    (function() {
        const colorSchemeKey = 'StackColorScheme';
        const colorSchemeItem = localStorage.getItem(colorSchemeKey);
        const supportDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches === true;

        if (colorSchemeItem == 'dark' || colorSchemeItem === 'auto' && supportDarkMode) {
            

            document.documentElement.dataset.scheme = 'dark';
        } else {
            document.documentElement.dataset.scheme = 'light';
        }
    })();
</script>
<div class="container main-container flex on-phone--column extended"><aside class="sidebar left-sidebar sticky ">
    <button class="hamburger hamburger--spin" type="button" id="toggle-menu" aria-label="Toggle Menu">
        <span class="hamburger-box">
            <span class="hamburger-inner"></span>
        </span>
    </button>

    <header>
        
            
            <figure class="site-avatar">
                <a href="/">
                
                    
                    
                    
                        
                        <img src="/img/profile_hu_d0d4f9ff9a7faeba.jpg" width="300"
                            height="299" class="site-logo" loading="lazy" alt="Avatar">
                    
                
                </a>
                
            </figure>
            
        
        
        <div class="site-meta">
            <h1 class="site-name"><a href="/">Xuan&#39;s Blog</a></h1>
            <h2 class="site-description">長大以後就是小瑄</h2>
        </div>
    </header><ol class="menu-social">
            
                <li>
                    <a 
                        href='https://github.com/pxchen110'
                        target="_blank"
                        title="GitHub"
                        rel="me"
                    >
                        
                        
                            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-brand-github" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M9 19c-4.3 1.4 -4.3 -2.5 -6 -3m12 5v-3.5c0 -1 .1 -1.4 -.5 -2c2.8 -.3 5.5 -1.4 5.5 -6a4.6 4.6 0 0 0 -1.3 -3.2a4.2 4.2 0 0 0 -.1 -3.2s-1.1 -.3 -3.5 1.3a12.3 12.3 0 0 0 -6.2 0c-2.4 -1.6 -3.5 -1.3 -3.5 -1.3a4.2 4.2 0 0 0 -.1 3.2a4.6 4.6 0 0 0 -1.3 3.2c0 4.6 2.7 5.7 5.5 6c-.6 .6 -.6 1.2 -.5 2v3.5" />
</svg>



                        
                    </a>
                </li>
            
        </ol><ol class="menu" id="main-menu">
        
        
        
        <li >
            <a href='/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-home" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <polyline points="5 12 3 12 12 3 21 12 19 12" />
  <path d="M5 12v7a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-7" />
  <path d="M9 21v-6a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v6" />
</svg>



                
                <span>Home</span>
            </a>
        </li>
        
        
        <li >
            <a href='/about/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-user" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="7" r="4" />
  <path d="M6 21v-2a4 4 0 0 1 4 -4h4a4 4 0 0 1 4 4v2" />
</svg>



                
                <span>About</span>
            </a>
        </li>
        
        
        <li >
            <a href='/archives/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-archive" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <rect x="3" y="4" width="18" height="4" rx="2" />
  <path d="M5 8v10a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-10" />
  <line x1="10" y1="12" x2="14" y2="12" />
</svg>



                
                <span>Archives</span>
            </a>
        </li>
        
        
        <li >
            <a href='/search/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-search" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="10" cy="10" r="7" />
  <line x1="21" y1="21" x2="15" y2="15" />
</svg>



                
                <span>Search</span>
            </a>
        </li>
        
        
        <li >
            <a href='/links/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-link" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <path d="M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5" />
  <path d="M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5" />
</svg>



                
                <span>Links</span>
            </a>
        </li>
        
        <li class="menu-bottom-section">
            <ol class="menu">

                
                    <li id="dark-mode-toggle">
                        <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-toggle-left" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="8" cy="12" r="2" />
  <rect x="2" y="6" width="20" height="12" rx="6" />
</svg>



                        <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-toggle-right" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="16" cy="12" r="2" />
  <rect x="2" y="6" width="20" height="12" rx="6" />
</svg>



                        <span>Dark Mode</span>
                    </li>
                
            </ol>
        </li>
    </ol>
</aside>

    <aside class="sidebar right-sidebar sticky">
        
            
                
    <section class="widget archives">
        <div class="widget-icon">
            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-hash" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <line x1="5" y1="9" x2="19" y2="9" />
  <line x1="5" y1="15" x2="19" y2="15" />
  <line x1="11" y1="4" x2="7" y2="20" />
  <line x1="17" y1="4" x2="13" y2="20" />
</svg>



        </div>
        <h2 class="widget-title section-title">Table of contents</h2>
        
        <div class="widget--toc">
            <nav id="TableOfContents">
  <ol>
    <li><a href="#樣本收集">樣本收集</a></li>
    <li><a href="#v4-與-v3-差異">v4 與 v3 差異</a>
      <ol>
        <li><a href="#取消-subset-釋出">取消 subset 釋出</a></li>
        <li><a href="#filtering-allele-frequency-faf">filtering allele frequency (FAF)</a></li>
      </ol>
    </li>
    <li><a href="#v4-exome-分析流程">v4 exome 分析流程</a></li>
    <li><a href="#v4-sv">v4 SV</a></li>
    <li><a href="#v4-cnv">v4 CNV</a>
      <ol>
        <li><a href="#cnv-註釋">CNV 註釋</a></li>
        <li><a href="#cnv-subsets">CNV subsets</a></li>
      </ol>
    </li>
    <li><a href="#references">References</a></li>
  </ol>
</nav>
        </div>
    </section>

            
        
    </aside>


            <main class="main full-width">
    <article class="has-image main-article">
    <header class="article-header">
        <div class="article-image">
            <a href="/p/%E7%94%9F%E8%B3%87%E7%B3%BB%E5%88%97-gnomad-v3-database-%E4%BB%8B%E7%B4%B9/">
                
                    <img src="/post/bioinfo-gnomad-v3/20240516_img1.png" loading="lazy" alt="Featured image of post 生資系列： gnomAD v3 database 介紹" />
                
            </a>
        </div>
    

    <div class="article-details">
    
    <header class="article-category">
        
            <a href="/categories/bioinformatics/" >
                Bioinformatics
            </a>
        
    </header>
    

    <div class="article-title-wrapper">
        <h2 class="article-title">
            <a href="/p/%E7%94%9F%E8%B3%87%E7%B3%BB%E5%88%97-gnomad-v3-database-%E4%BB%8B%E7%B4%B9/">生資系列： gnomAD v3 database 介紹</a>
        </h2>
    
        
    </div>

    
    
    
    
    <footer class="article-time">
        
            <div>
                <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-calendar-time" width="56" height="56" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <path d="M11.795 21h-6.795a2 2 0 0 1 -2 -2v-12a2 2 0 0 1 2 -2h12a2 2 0 0 1 2 2v4" />
  <circle cx="18" cy="18" r="4" />
  <path d="M15 3v4" />
  <path d="M7 3v4" />
  <path d="M3 11h16" />
  <path d="M18 16.496v1.504l1 1" />
</svg>
                <time class="article-time--published">May 16, 2024</time>
            </div>
        

        
            <div>
                <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-clock" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <polyline points="12 7 12 12 15 15" />
</svg>



                <time class="article-time--reading">
                    3 minute read
                </time>
            </div>
        
    </footer>
    

    
</div>

</header>

    <section class="article-content">
    
    
    <p>(Image from <a class="link" href="https://gnomad.broadinstitute.org/"  target="_blank" rel="noopener"
    >gnomAD</a> website)</p>
<p>上一篇文章著重討論 The Genome Aggregation Database (gnomAD) v3 版內容介紹，本文將介紹去年11月更新之<strong>gnomAD v4</strong> (以及部分今年4月更新之 <strong>v4.1</strong>)。</p>
<!-- more -->
<!-- toc -->
<h2 id="樣本收集">樣本收集
</h2><p>下表整理不同版本間使用的樣本數與參考序列，最大的更新為: <strong>v4 增收 416,555 UK Biobank WES data</strong></p>
<div class="table-wrapper"><table>
  <thead>
      <tr>
          <th style="text-align: left">version</th>
          <th style="text-align: left">reference</th>
          <th style="text-align: left">(type)sample size</th>
      </tr>
  </thead>
  <tbody>
      <tr>
          <td style="text-align: left">gnomAD v2.1</td>
          <td style="text-align: left">GRCh37</td>
          <td style="text-align: left">(WES) 125,748 ; (WGS) 15,708</td>
      </tr>
      <tr>
          <td style="text-align: left">gnomAD-SV v2.1</td>
          <td style="text-align: left">GRCh37</td>
          <td style="text-align: left">(WGS) 10,847</td>
      </tr>
      <tr>
          <td style="text-align: left">gnomAD v3.1</td>
          <td style="text-align: left">GRCh38</td>
          <td style="text-align: left">(WGS) 76,156</td>
      </tr>
      <tr>
          <td style="text-align: left">gnomAD v4.0</td>
          <td style="text-align: left">GRCh38</td>
          <td style="text-align: left">(WES) 730,947 ; (WGS) 76,215</td>
      </tr>
      <tr>
          <td style="text-align: left">gnomAD-SV v4.0</td>
          <td style="text-align: left">GRCh38</td>
          <td style="text-align: left">(WGS) 63,046</td>
      </tr>
      <tr>
          <td style="text-align: left">gnomAD-CNV v4.0</td>
          <td style="text-align: left">GRCh38</td>
          <td style="text-align: left">(WES) 464,297</td>
      </tr>
  </tbody>
</table></div>
<p>從官網的統計圖表可清楚了解族群數量分布，由於UK Biobank 樣本以 European 為大宗，合併至 v4 使 European Ancestry比例偏高接近80%
另外值得注意的是 EAS 樣本數有大進展，從前幾版的不到萬人增至2萬初，能協助修正 allele frequency 的預估結果</p>
<img src="/post/bioinfo-gnomad-v4/20240523_img2.jpg"  width="600">
<h2 id="v4-與-v3-差異">v4 與 v3 差異
</h2><h3 id="取消-subset-釋出">取消 subset 釋出
</h3><p>過去v3樣本來自多種cohort，為提供研究者作為疾病研究的control，gnomAD v3 有釋出多種subsets，如: non-cancer subset 為排除罹患癌症或從相關 cohort study收集來的樣本群(更多介紹可參考前篇文章)。
<strong>在v4版本僅有釋出non-UKBB subset</strong>，且僅提供allele frequency資訊給研究者做累加使用。下面文字截自gnomAD v4 SNV chrY VCF檔的INFO field，可以看到<strong>每個population都有拆分另外一組non-UKB的(prefix: non_ukb) 數據:</strong></p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">AC_afr_XY=0;AN_afr_XY=0;nhomalt_afr_XY=0;AC_afr=0;AN_afr=0;nhomalt_afr=0;...
</span></span><span class="line"><span class="cl">AC_eas_XY=0;AN_eas_XY=0;nhomalt_eas_XY=0;...
</span></span><span class="line"><span class="cl">AC_non_ukb_afr_XY=0;AN_non_ukb_afr_XY=0;nhomalt_non_ukb_afr_XY=0;AC_non_ukb_afr=0;AN_non_ukb_afr=0;nhomalt_non_ukb_afr=0;...
</span></span><span class="line"><span class="cl">AC_non_ukb_eas_XY=0;AN_non_ukb_eas_XY=0;nhomalt_non_ukb_eas_XY=0;AC_non_ukb_eas=0;AN_non_ukb_eas=0;nhomalt_non_ukb_eas=0;...
</span></span></code></pre></td></tr></table>
</div>
</div><p>此外若有使用註解 gnomAD 資料庫的工具如 dbnsfp，則可以明顯看到在資料庫版本更新後，就不會有  <code>gnomAD_exomes_non_cancer</code> 類似欄位產生，只會剩下 <code>gnomAD_exomes_non_ukb</code> 等欄位</p>
<p>而v4取消製作subset原因包含:</p>
<ul>
<li>研究特定疾病如cancer、neuro-disease等，這些疾病的盛行率並不會過高到需要透過(乾淨的?) subsets 來做比對與分析，建議直接使用 gnomAD metadata 來協助研究</li>
<li>新加入的 UK Biobank 或是部分 cohort samples 並沒有完整 phenotype 資料，無法明確的區分樣本的疾病狀態</li>
</ul>
<h3 id="filtering-allele-frequency-faf">filtering allele frequency (FAF)
</h3><p>將 WES 和 WGS 的 allele count 合併計算之 allele frequency，稱作 filtering allele frequency</p>
<h2 id="v4-exome-分析流程">v4 exome 分析流程
</h2><p>(WGS 流程可參照 gnomAD v3版作法)
由於 v4 最大進展在於 exome callset 樣本數的提升，因此後續分析流程也有些更新:</p>
<ul>
<li>joint callset 由原本的 gVCF 格式轉為 VariantDataset (VDS) 格式儲存</li>
<li>增加 <strong>interval QC</strong> 以確保目標區段擁有足夠的<strong>深度</strong>判別變異: 官網提供一組 <a class="link" href="https://gnomad.broadinstitute.org/help/what-are-your-calling-intervals-what-intervals-are-used-for-the-genomes-coding-only-files"  target="_blank" rel="noopener"
    >interval list</a>，左右延伸(padding) 50bp 後計算每個樣本在這些區段的平均深度。最後將 &gt;85% 樣本擁有 &gt;20X 深度的區段定義為 high coverage intervals</li>
<li>為 sample QC 定義 <strong>high qulaity site</strong>:為判斷樣本族群來源、親緣關係或定序品質，gnomAD v4從幾個資料集挑選 high quality sites (n = 175,043) 協助樣本QC:
<ul>
<li>屬於 autosomal, biallelic, single nucleotide variants (SNVs)</li>
<li>位於 gnomAD v3, CCDG, 以及 UK Biobank dataset 的位點紀錄</li>
</ul>
</li>
<li>sample QC: 使用上面這些high quality site進行以下幾個重點QC
<ul>
<li>hard filters: 使用新工具從 genotype data 預估 contamination 而非重新拿raw read 分析。其他也用DP, Het/Hom ratio 等數值去篩</li>
<li>PCA clustering for exome capture platform: 由於樣本來源多樣，使用PCA區分出20種定序平台，label上潛在差異</li>
<li>sex karyotypes</li>
<li>relatedness between v4 exomes and genomes, and identified pairs of first and second degree relatives</li>
<li>genetic ancestry groups</li>
<li>QC metrics</li>
</ul>
</li>
<li>variant QC by VQSR (和 v3 一樣拿大型計劃的位點做模型訓練並應用類似的篩選條件)</li>
</ul>
<hr>
<h2 id="v4-sv">v4 SV
</h2><p>和前版 v2.1相比，除了<strong>樣本數</strong>從一萬初<strong>增加至超過六萬</strong>、偵測位點數增加，參考序列也從 hg19/GRCh37 改成擁有 alternative contigs 的 <strong>GRCh38做為參考序列</strong>。
偵測流程使用相同的 GATK-SV pipeline，並增進變異註解(annotation)的結果。gnomAD-SV v4 從63,046個WGS樣本中找到超過一百萬個 high-quality SVs，SV特徵的趨勢與前版本相同，多為短片段(median size 360bp)的罕見位點(96% SVs AF &lt; 1%)。</p>
<p>gnomAD-SV v4 的結果已公開至網站，可先行查詢利用；至於技術細節等內容，等研究論文發表後再來更新，根據之前使用v2.1版資料的經驗，可以再拿新的East Asian AF 來做 correlation plot，不過中間需要針對變異座標進行liftover轉換，等有空再來嘗試看看。</p>
<h2 id="v4-cnv">v4 CNV
</h2><p><strong>gnomAD v4第一次釋出 CNV callset</strong>。也代表未來關於致病位點的研究，可往大片段缺失的偵測目標前進，並擁有 general population 資訊可參考。目前收錄從 464,297 WES data 找到之 rare ( AF &lt;1% ) autosomal coding copy number variants (CNVs)，使用 GATK-gCNV (Babadi et al. Nat Genet, 2023) 流程偵測變異。</p>
<p>簡單順一下分析流程，主要利用比對至參考序列的BAM/CRAM作為input，並計算每個 protein-coding region (target interval) 的 read count (CNV 主要以深度差異來判斷) 接著使用這些數值做 clustering ，將特徵接近的樣本歸類至同一批次(batch)做後續分析。</p>
<p>CNV流程需要baseline來比較深度差異，通常來自 panel of normal (PoN，在somatic偵測代表tumor相對應的正常樣本) 的結果，而gnomAD 無針對特殊疾病，所以採隨機挑選200樣本作為Pon，之後經由QC篩選，normalization等等，最終以read depth, read count 等特徵判定基因體內的CNV events。</p>
<p>CNV偵測流程的可信度，透過一組同時具有WES(default input), WGS (IGV validation) 和 microarray 的樣本來驗證，透過和不同開源工具比較，詳細工具的演算法與使用差異，有機會再細看比較。</p>
<h3 id="cnv-註釋">CNV 註釋
</h3><p>在變異註釋方面，現今工具主要在短片段如SNV/indel方面詮釋較佳，如ensembl VEP, AnnotSv等，而CNV宜由於變異跨度大，對於現行臨床ACMG準則來說難以完整判定，且多以基因為單位個別做註釋，整體影響預測尚未完全，而gnomAD作為可能的reference dataset，也針對變異作了以下註釋:</p>
<ul>
<li>基本的GENCODE 註釋，並以overlap的比例判斷是否給註釋</li>
<li>長度與exon, interval範圍跨度</li>
<li>集合與分族群統計的AF: site count (SC), site number (SN), and site frequency (SF)</li>
</ul>
<h3 id="cnv-subsets">CNV subsets
</h3><p>和 gnomADv4 不同，由於大片段的罕見CNV位點與精神疾病有關連性，<strong>gnomAD CNV 有釋出兩組subsets</strong>:</p>
<blockquote>
<p>Due to the known enrichment of large, rare CNVs that confer substantial relative risk in neuropsychiatric phenotype
neuropsychiatric conditions: bipolar, epilepsy, schizophrenia, psychosis, and general psychiatric conditions</p></blockquote>
<ul>
<li>non-neuro (n = 418,924): 非精神疾病相關cohort樣本(neuropsychiatric conditions)，以及 biobank 樣本</li>
<li>non-neuro-control(n = 269,919): 不同cohort內標示為control的樣本,或是biobank樣本</li>
</ul>
<h2 id="references">References
</h2><p><a class="link" href="https://gnomad.broadinstitute.org/news/2023-11-gnomad-v4-0/"  target="_blank" rel="noopener"
    >https://gnomad.broadinstitute.org/news/2023-11-gnomad-v4-0/</a>
<a class="link" href="https://gnomad.broadinstitute.org/news/2023-11-v4-structural-variants/"  target="_blank" rel="noopener"
    >https://gnomad.broadinstitute.org/news/2023-11-v4-structural-variants/</a>
<a class="link" href="https://gnomad.broadinstitute.org/news/2023-11-v4-copy-number-variants/"  target="_blank" rel="noopener"
    >https://gnomad.broadinstitute.org/news/2023-11-v4-copy-number-variants/</a>
<a class="link" href="https://gnomad.broadinstitute.org/news/2019-10-gnomad-v3-0/"  target="_blank" rel="noopener"
    >https://gnomad.broadinstitute.org/news/2019-10-gnomad-v3-0/</a></p>

</section>


    <footer class="article-footer">
    
    <section class="article-tags">
        
            <a href="/tags/bioinformatics/">Bioinformatics</a>
        
            <a href="/tags/database/">Database</a>
        
    </section>


    
    <section class="article-copyright">
        <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-copyright" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <path d="M14.5 9a3.5 4 0 1 0 0 6" />
</svg>



        <span>Licensed under CC BY-NC-SA 4.0</span>
    </section>
    </footer>


    
</article>

    

    

<aside class="related-content--wrapper">
    <h2 class="section-title">Related content</h2>
    <div class="related-content">
        <div class="flex article-list--tile">
            
                
<article class="has-image">
    <a href="/p/%E7%94%9F%E8%B3%87%E7%B3%BB%E5%88%97-%E4%B8%8B%E8%BC%89ncbi%E8%B3%87%E6%96%99%E7%9A%84%E6%96%B9%E6%B3%95/">
        
        
            <div class="article-image">
                
                    <img src="/post/bioinfo-ncbi/20240829_img1.png" loading="lazy" data-key="" data-hash="/post/bioinfo-ncbi/20240829_img1.png"/>
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">生資系列： 下載NCBI資料的方法</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/p/%E7%94%9F%E8%B3%87%E7%B3%BB%E5%88%97%E7%A2%BA%E6%A0%A1%E8%B3%87%E6%96%99%E9%9B%86-benchmarking-truth-set/">
        
        
            <div class="article-image">
                
                    <img src="/post/bioinfo-benchmarking/20240606_img0.png" loading="lazy" data-key="" data-hash="/post/bioinfo-benchmarking/20240606_img0.png"/>
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">生資系列：確校資料集 Benchmarking truth set</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/p/%E7%94%9F%E8%B3%87%E7%B3%BB%E5%88%97-gnomad-v3-database-%E4%BB%8B%E7%B4%B9/">
        
        
            <div class="article-image">
                
                    <img src="/post/bioinfo-gnomad-v4/20240523_img1.png" loading="lazy" data-key="" data-hash="/post/bioinfo-gnomad-v4/20240523_img1.png"/>
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">生資系列： gnomAD v3 database 介紹</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/p/%E7%94%9F%E8%B3%87%E7%B3%BB%E5%88%97-all-of-us-research-program-%E4%BB%8B%E7%B4%B9/">
        
        
            <div class="article-image">
                
                    <img src="/post/bioinfo-allofus/20240509_img1.jpg" loading="lazy" data-key="" data-hash="/post/bioinfo-allofus/20240509_img1.jpg"/>
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">生資系列： All of US research program 介紹</h2>
        </div>
    </a>
</article>

            
                
<article class="">
    <a href="/p/%E7%94%9F%E8%B3%87%E7%B3%BB%E5%88%97%E5%9F%BA%E5%9B%A0%E8%A8%BB%E8%A7%A3%E8%BD%89%E6%8F%9B%E5%A5%97%E4%BB%B6-biomart-bitr-mygene/">
        
        

        <div class="article-details">
            <h2 class="article-title">生資系列：基因註解轉換套件 biomaRt, bitr, mygene </h2>
        </div>
    </a>
</article>

            
        </div>
    </div>
</aside>

     
    
        
    <div class="disqus-container">
    <div id="disqus_thread"></div>
<script>
    window.disqus_config = function () {
    
    
    
    };
    (function() {
        if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
            document.getElementById('disqus_thread').innerHTML = 'Disqus comments not available by default when the website is previewed locally.';
            return;
        }
        var d = document, s = d.createElement('script'); s.async = true;
        s.src = '//' + "pxchen110-blog" + '.disqus.com/embed.js';
        s.setAttribute('data-timestamp', +new Date());
        (d.head || d.body).appendChild(s);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
</div>

<style>
    .disqus-container {
        background-color: var(--card-background);
        border-radius: var(--card-border-radius);
        box-shadow: var(--shadow-l1);
        padding: var(--card-padding);
    }
</style>

<script>
    window.addEventListener('onColorSchemeChange', (e) => {
        if (typeof DISQUS == 'object') {
            DISQUS.reset({
                reload: true
            });
        }
    })
</script>

    

    <footer class="site-footer">
    <section class="copyright">
        &copy; 
        
            2024 - 
        
        2025 pxchen

    </section>
    
    <section class="powerby">
        Built with <a href="https://gohugo.io/" target="_blank" rel="noopener">Hugo</a> <br />
        Theme <b><a href="https://github.com/CaiJimmy/hugo-theme-stack" target="_blank" rel="noopener" data-version="3.30.0">Stack</a></b> designed by <a href="https://jimmycai.com" target="_blank" rel="noopener">Jimmy</a>
    </section>
</footer>


    
<div class="pswp" tabindex="-1" role="dialog" aria-hidden="true">

    
    <div class="pswp__bg"></div>

    
    <div class="pswp__scroll-wrap">

        
        <div class="pswp__container">
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
        </div>

        
        <div class="pswp__ui pswp__ui--hidden">

            <div class="pswp__top-bar">

                

                <div class="pswp__counter"></div>

                <button class="pswp__button pswp__button--close" title="Close (Esc)"></button>

                <button class="pswp__button pswp__button--share" title="Share"></button>

                <button class="pswp__button pswp__button--fs" title="Toggle fullscreen"></button>

                <button class="pswp__button pswp__button--zoom" title="Zoom in/out"></button>

                
                
                <div class="pswp__preloader">
                    <div class="pswp__preloader__icn">
                        <div class="pswp__preloader__cut">
                            <div class="pswp__preloader__donut"></div>
                        </div>
                    </div>
                </div>
            </div>

            <div class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap">
                <div class="pswp__share-tooltip"></div>
            </div>

            <button class="pswp__button pswp__button--arrow--left" title="Previous (arrow left)">
            </button>

            <button class="pswp__button pswp__button--arrow--right" title="Next (arrow right)">
            </button>

            <div class="pswp__caption">
                <div class="pswp__caption__center"></div>
            </div>

        </div>

    </div>

</div><script 
                src="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe.min.js"integrity="sha256-ePwmChbbvXbsO02lbM3HoHbSHTHFAeChekF1xKJdleo="crossorigin="anonymous"
                defer
                >
            </script><script 
                src="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe-ui-default.min.js"integrity="sha256-UKkzOn/w1mBxRmLLGrSeyB4e1xbrp4xylgAWb3M42pU="crossorigin="anonymous"
                defer
                >
            </script><link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/default-skin/default-skin.min.css"crossorigin="anonymous"
            ><link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe.min.css"crossorigin="anonymous"
            >

            </main>
        </div>
        <script 
                src="https://cdn.jsdelivr.net/npm/node-vibrant@3.1.6/dist/vibrant.min.js"integrity="sha256-awcR2jno4kI5X0zL8ex0vi2z&#43;KMkF24hUW8WePSA9HM="crossorigin="anonymous"
                
                >
            </script><script type="text/javascript" src="/ts/main.1e9a3bafd846ced4c345d084b355fb8c7bae75701c338f8a1f8a82c780137826.js" defer></script>
<script>
    (function () {
        const customFont = document.createElement('link');
        customFont.href = "https://fonts.googleapis.com/css2?family=Lato:wght@300;400;700&display=swap";

        customFont.type = "text/css";
        customFont.rel = "stylesheet";

        document.head.appendChild(customFont);
    }());
</script>

    </body>
</html>
